### DISCUSSION TITLE
Latest version now self prompts?

### DISCUSSION THREAD

**Discussion by ghost** [oobabooga/text-generation-webui#2059] [General]
Just did an update now as it completes my prompt it self-prompts with "human: bla bla" and takes off on its own ...  Same settings and model as before so not sure what happened here?

**Comment by m-spangenberg** [comment]
I'm also experiencing self prompting.

---

Quickly tested: Did a git pull and updated the requirements, `GPT4All-13B-snoozy.ggml.q4_0` self-prompts in default mode, but not in chat mode. `WizardLM-7B-uncensored.ggml.q5_1` does not self-prompt in either mode. Which model are you using?

**Reply by ghost** [reply]
After some more playing last night, if i changed the prompt settings it started working again. Stable Vicuna prompt was working before, but i switched to one of the 'base' vicuna prompts and the self prompt went away. ( both 1.0 and 1.1 seemed to do the same for me ).  I also normally use the llama-precise general settings or i get lots of garbage results.  ( stable-vicuna-13B.ggml.q4_0 ) . i was using instruct mode. But saw it in chat mode too.

( i might have the wording wrong on the prompts for exact wording, i cant go look since its dead.. see next comment )

Went back into it today to do some more testing and document things better, but i did an update first and i guess that was a bad idea. Broke now. So going to take some time after bit to start over fresh. 

Getting following now on start:  

ile "server.py", line 56
    yield f"The settings for {selected_model} have been updated.\nClick on \"Load the model\" to load it."

**Reply by m-spangenberg** [reply]
Strange, maybe try starting the server using `--model-menu` flag?

---

If you're on Linux and you're comfortable in the command-line, virtual environments are the way to go if you want to test changes without breaking stuff. For CPU use at least, the steps below work really well for me. Plus, it caches downloads, so you don't sit around waiting every time you make a new environment.

```
# grab the repo
git clone https://github.com/oobabooga/text-generation-webui.git && cd text-generation-webui/

# enter the venv shell
pipenv shell

# install the requirements
pip install -r requirements.txt

# start the server
python3 server.py --cpu --model-menu --chat
```

**Reply by ghost** [reply]
Ill try the model-menu flag when i get back to my desk this afternoon. Never used that before so never thought of it..  I do use pipenv, but when you update within the environment, sort of hard to go back :)   ya, i know i should be doing local source control or something.. but i'm lazy and this isn't production, so i can afford to spend time fixing.  ( i mostly use it as a 'coding aid' )

-edit the menu parameter didnt work, nor does specifying the model. Same message as above. Tonight ill just start fresh and see what happens.  Not like anything will be lost by starting over.

-edit 2.. meeting canceled had a few mins. Tore the environment out and rebuilt. Now the interface is loading, however i guess my current model format is too old, will have to DL a newer one later. But it starts again ok. Something must have gone sideways during the update.

**Comment by dblacknc** [comment]
I've found adding the string that occurs as the model starts self-prompting to the custom stopping strings on the Parameters tab seems to help.  Basically when the model *outputs* something like "\nHuman:" - or whatever the prompt tag is, that becomes another trigger for generation to stop.

**Reply by ghost** [reply]
Never noticed that option, ill keep that in mind if it returns.

**Comment by ghost** [comment]
- Rebuilding the environment did fix the fail to load problem. who knows what happened there. Something got out of whack during the pull for the update. 
- Updated my model to a newer version, since what i had was no longer supported ( now im at 5.1, via the bloke's repo ). I'm sure i could have converted it myself but its faster just to download a new one.
- Using the stable V prompt like i used to I still get the self-prompting.  Switched to stock vicuna 1.1, still had self- prompt, both instruct and chat instruct. Didn't try 'just' chat.  But going to V0 prompt settings, it works. 

Its weird things changed in this way, and how its working, BUT its working.. so good enough for me :)
